### PR TITLE
Feature/basic api

### PR DESCRIPTION
**Pull request by norling**
Min idé om grundläggande struktur om vi kör flask/peewee för APIet.
Om det ser ok ut så börjar jag kolla på OpenID lite mer, och dan kan kolla på att göra grundstrukturen för en react-sida.

### CODE REVIEW

File: app/test.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    def setUp(self):
+        db.set_database("herdbook", "localhost", "5432", "herdbook", "insecure")
+
+    def test_genebank(self):
+        """
+        Checks that the genebank table exists and that all the fields in the
+        mapping exists in the database.
+        """
+        self.assertTrue(db.Genebank.table_exists())
+        db.Genebank.select().execute(db.DB_PROXY)
```
Comment (jonandernovella) [inline app/test.py]: Could we make sure that the models are actually bound to the given DB? I.e: `assert Class._meta.database is db.DB_PROXY`

**Comment by pontus** [comment]
Looks fine for the moment as far as I'm concerned.

In the future (and could be done now) I think we should move to generate the models with pwiz.

**Comment by norling** [comment]
pontus - are you suggesting that the model should be generated on the fly from the database? or just that it should be generated automatically instead of manually?

**Comment by pontus** [comment]
Not on the fly, just automatic.

**Comment by andkaha** [comment]
Hold on, why do you remove the schema file?! How do I create a database to load the data from CSV now? Or did you also write a loading script?

**Comment by norling** [comment]
With python, since we weren't supposed to have both SQL and python.

**Comment by andkaha** [comment]
Hm. What makes it impossible to have both SQL and Python?

**Comment by norling** [comment]
Nothing. But this PR had a change request where the database shouldn't be defined in both python and SQL... The first idea was to auto-generate the python-mapping from the SQL, but that didn't generate the python docstrings, so then I went with generating the SQL from python.

File: app/herdbook.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+"""
+This file contains the main webserver of 'the herdbook'.
+
+The server uses Flask to serve a React frontend, connect to a postgres database,
+and do calculations with R.
```
Comment (jhagberg) [inline app/herdbook.py]: We are not sure about R yet

Comment (jhagberg) [reply app/herdbook.py]: Not a big issue

File: app/herdbook.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            },
+        'father': {
+            'id': individual.father.id,
+            'name': individual.father.name
+            },
+        'colour': individual.colour.name,
+        'colour_note': individual.colour_note,
+        'death_date': individual.death_date,
+        'death_note': individual.death_note,
+        'weight_young': individual.weight_young,
```
Comment (jhagberg) [inline app/herdbook.py:60]: Varför heter det weight_young?

File: app/herdbook.py (hunk 2 above)
Comment (norling) [reply app/herdbook.py]: true, I can update the comments!

File: app/herdbook.py (hunk 3 above)
Comment (jhagberg) [reply app/herdbook.py:60]: This data is also loaded in weight table according to @kusalananda

File: app/test.py (hunk 1 above)
Comment (jhagberg) [reply app/test.py]: Is this relevant still?

**Pull request merged**